src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -76,51 +76,80 @@ class WorkspaceNewRoomPage extends React.Component {
         this.setState({workspaceOptions: _.map(workspaces, policy => ({label: policy.name, key: policy.id, value: policy.id}))});
     }
 
+    validateAndCreatePolicyRoom() {
+        if (!this.validate()) {
+            return;
+        }
+        Report.createPolicyRoom(
+            this.state.policyID,
+            this.state.roomName,
+            this.state.visibility,
+        );
+    }
+
     /**
-     * Called when a workspace is selected. Also calls checkAndModifyRoomName,
-     * which displays an error if the given roomName exists on the newly selected workspace.
-     * @param {String} policyID
+     * @returns {Boolean}
      */
-    onWorkspaceSelect(policyID) {
-        this.setState({policyID});
-        this.checkAndModifyRoomName(this.state.roomName);
+    validate() {
+        const errors = {};
+
+        const isExistingRoomName = _.some(
+            _.values(this.props.reports),
+            report => report && report.policyID === this.state.policyID
+            && report.reportName === this.state.roomName,
+        );
+
+        if (!this.state.roomName || this.state.roomName === CONST.ROOM_PREFIX) {
+            errors.roomName = this.props.translate('newRoomPage.pleaseEnterRoomName');
+        } else if (this.state.policyID && isExistingRoomName) {
+            errors.roomName = this.props.translate('newRoomPage.roomAlreadyExists');
+        }
+
+        if (!this.state.policyID) {
+            errors.policyID = this.props.translate('newRoomPage.pleaseChooseWorkspace');
+        }
+
+        if (!this.state.visibility) {
+            errors.visibility = this.props.translate('newRoomPage.pleaseChooseVisibility');
+        }
+
+        this.setState({errors});
+        return Boolean(_.isEmpty(errors));
```

Comment:
```suggestion
        return _.isEmpty(errors);
```

`_.isEmpty()` returns a Boolean so there is no need for the Boolean cast.

Reply:
Ok I will change it.